lib/Factory/ScheduleFactory.php

```diff
@@ -486,8 +486,91 @@ public function query($sortOrder = null, $filterBy = [])
         // End both dates
 
         if ($parsedFilter->getIntArray('displayGroupIds') != null) {
-            $body .= ' AND `schedule`.eventId IN (SELECT `lkscheduledisplaygroup`.eventId FROM `lkscheduledisplaygroup`
-             WHERE displayGroupId IN (' . implode(',', $parsedFilter->getIntArray('displayGroupIds')) . ')) ';
+            // non Schedule grid filter, keep it the way it was.
+            if ($parsedFilter->getInt('sharedSchedule') === null &&
+                $parsedFilter->getInt('directSchedule') === null
+            ) {
+                $body .= ' AND `schedule`.eventId IN (
+                    SELECT `lkscheduledisplaygroup`.eventId FROM `lkscheduledisplaygroup`
+                    WHERE displayGroupId IN (' . implode(',', $parsedFilter->getIntArray('displayGroupIds')) . ')
+                     ) ';
+            } else {
+                // Schedule grid query
+                // check what options we were provided with and adjust query accordingly.
+                if ($parsedFilter->getInt('sharedSchedule') === 1) {
```

Comment:
Can we add a couple of comments to each of these conditions to describe what the SQL is supposed to do?

Reply:
I've changed the structure ever so slightly ;) and the sql a little bit as well.

Comments for each section as there now too, which should help, thanks